Source/WebCore/page/Navigation.cpp

```diff
@@ -147,6 +147,49 @@ void Navigation::updateForActivation(HistoryItem* previousItem, std::optional<Na
     m_activation = NavigationActivation::create(*type, *currentEntry(), WTFMove(previousEntry));
 }
 
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#fire-the-pageswap-event
+RefPtr<NavigationActivation> Navigation::createForPageswapEvent(HistoryItem* newItem, DocumentLoader* documentLoader)
+{
+    auto type = documentLoader->triggeringAction().navigationAPIType();
+    if (!type)
+        return nullptr;
+
+    bool isSameOrigin = SecurityOrigin::create(documentLoader->documentURL())->isSameOriginAs(window()->protectedDocument()->securityOrigin());
+    if (!isSameOrigin || !documentLoader->request().isSameSite())
+        return nullptr;
+
+    // The lookups into m_entries should succeed if the Navigation API is enabled, but
+    // we fallback to a new NavigationHistoryEntry if not so that the pageswap event can
```

Comment:
Maybe make this more explicit, check if NavigationAPI is enabled and do the "right thing", then fallback otherwise.

Reply:
Done!